Examples/TodoApp/TodoAppTests/src/viewmodel/TaskListViewModelTests.swift

```diff
@@ -0,0 +1,91 @@
+import XCTest
+@testable import TodoApp
+import MobiusCore
+
+final class TaskListViewModelTests: XCTestCase {
+
+    private var subject: TasksListViewModel!
+
+    private var effectHandler: TasksListEffectHandler!
+    private var mockTasksDataSource: MockTasksDataSource!
+    private var mockEventConsumer: MockEventConsumer!
+
+    override func setUp() {
+        super.setUp()
+        effectHandler = TasksListEffectHandler()
+        mockTasksDataSource = MockTasksDataSource()
+        mockEventConsumer = MockEventConsumer()
+        subject = TasksListViewModel(effectHandler: effectHandler,
+                                    dataSource: mockTasksDataSource,
+                                    eventConsumer: mockEventConsumer.mockConsumer)
+    }
+
+    override func tearDown() {
+        subject = nil
+        effectHandler = nil
+        mockTasksDataSource = nil
+        mockEventConsumer = nil
+        super.tearDown()
+    }
+
+    // MARK: - Tests
+
+    func test_viewWillAppear_sendsRefreshRequestedEvent() {
+        subject.viewWillAppear()
+        XCTAssertEqual(mockEventConsumer.eventSpy, .refreshRequested)
+    }
+
+    func test_didPressAddTaskButton_sendsRefreshRequestedEvent() {
+        subject.didPressAddTaskButton()
+        XCTAssertEqual(mockEventConsumer.eventSpy, .newTaskClicked)
+    }
+
+    func test_didCompleteTaskCreation_sendsTaskCreated() {
+        subject.didCompleteTaskCreation(title: "title", description: "description")
+        XCTAssertEqual(mockEventConsumer.eventSpy, .taskCreated(title: "title", description: "description"))
+    }
+}
+
+
+private final class MockTaskListEffectHandler: Connectable {
```

Comment:
Mocks will be moved to different files. and we could write more tests if we manage to inject the mobius loop properly, maybe using a factory. But we prefer to get some early feedback on this. More iterations to happen